scrapper/api/routes.py

```
@@ -0,0 +1,300 @@
+        if request.user_context:
+            user_context_dict = request.user_context.dict()
+        
+        # Execute 5-stage filtering pipeline
+        # Cap results at MAX_RESULT_LIMIT for clean UI
+        result_limit = request.limit_per_company
+        if result_limit is None:
+            result_limit = DEFAULT_RESULT_LIMIT
+        elif result_limit > MAX_RESULT_LIMIT:
+            result_limit = MAX_RESULT_LIMIT
+            logger.info(
+                f"Result limit capped at {MAX_RESULT_LIMIT} (requested: {request.limit_per_company})",
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)